### PR TITLE
feat: RPC lookup for Velas accounts

### PR DESCRIPTION
**Pull request by hrls**

### CODE REVIEW

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -46,6 +46,7 @@ solana-sdk = { path = "../sdk", version = "1.5.14" }
 solana-stake-program = { path = "../programs/stake", version = "1.5.14" }
 solana-vote-program = { path = "../programs/vote", version = "1.5.14" }
 solana-secp256k1-program = { path = "../programs/secp256k1", version = "1.5.14" }
+solana-account-decoder = { path = "../account-decoder", version = "1.5.14" }
```
Comment (vldm) [inline runtime/Cargo.toml]: i am not sure that this is a good idea, to depend runtime from highlevel crate. 
I am not even sure how this compile now, there should be cycle link?
Better put velas account inline part into runtime like with solana_spl token https://github.com/velas/velas-chain/blob/develop/runtime/src/inline_spl_token_v2_0.rs

File: validator/src/main.rs [hunk 2]
```diff
@@ -1578,7 +1585,10 @@ pub fn main() {
             "program-id" => AccountIndex::ProgramId,
             "spl-token-mint" => AccountIndex::SplTokenMint,
             "spl-token-owner" => AccountIndex::SplTokenOwner,
-            _ => unreachable!(),
+            "velas-accounts-storages" => AccountIndex::VelasAccountStorage,
+            "velas-accounts-owners" => AccountIndex::VelasAccountOwner,
+            "velas-accounts-operationals" => AccountIndex::VelasAccountOperational,
+            unexpected => panic!("Unable to handle 'account_indexes' flag {}", unexpected),
```
Comment (vldm) [inline validator/src/main.rs:1591]: :+1:

File: run.sh [hunk 3]
```diff
@@ -112,6 +112,10 @@ args=(
     --accounts-db-caching-enabled
     --snapshot-interval-slots 100
     --require-tower
+    --account-index program-id
+    --account-index velas-accounts-storages
+    --account-index velas-accounts-owners
+    --account-index velas-accounts-operationals
```
Comment (vldm) [inline run.sh:120]: Add rest of indexes or remove this after testing.

File: account-decoder/src/parse_velas_account.rs [hunk 4]
```diff
@@ -1,8 +1,29 @@
...
 use borsh::BorshDeserialize;
 use velas_account::*;
-const ACCOUNT_LEN: usize = 67;
+
+// TODO: try to avoid direct size hardcodes
+pub const ACCOUNT_LEN: usize = 67;
+
+// TODO: test it
+pub const ACCOUNT_STORAGE_KEY_OFFSET: usize =
+    ACCOUNT_LEN - std::mem::size_of::<solana_sdk::pubkey::Pubkey>();
```
Comment (vldm) [inline account-decoder/src/parse_velas_account.rs]: is it used somewhere?

Comment (hrls) [reply account-decoder/src/parse_velas_account.rs]: No.

File: run.sh (hunk 3 above)
Comment (hrls) [reply run.sh:120]: Okay, more is more!

File: runtime/Cargo.toml (hunk 1 above)
Comment (hrls) [reply runtime/Cargo.toml]: It works somehow now =)

Inlined SPL token declares simple stuff. But Velas Account needs full struct definition (or other *simple* way to find pubkey in Storage lists).

**Review by vldm — APPROVED** [review]

**Review by vldm — APPROVED** [review]

**Pull request merged**